modules/openideal_user/src/EventSubscriber/OpenidealUserEventSubscriber.php

```diff
@@ -85,4 +161,23 @@ public function stateChanged(ContentModerationStateChangedEvent $event) {
     $this->eventDispatcher->dispatch(OpenidealUserEvents::WORKFLOW_STATE_CHANGED, $event);
   }
 
+  /**
+   * Generate redirect response.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\FilterResponseEvent $event
+   *   Event.
+   */
+  protected function generateRedirectResponse(FilterResponseEvent $event) {
+    // @Todo: need to check for existing query params?
```

Comment:
Sure, let's do that.
We need to get the existing URL and append our GET parameter.

Reply:
Done